src/Transports.AspNetCore/IWebSocketHandler.cs

```diff
@@ -0,0 +1,27 @@
+using System.Net.WebSockets;
+using GraphQL.Types;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <inheritdoc/>
+public interface IWebSocketHandler<TSchema> : IWebSocketHandler
+    where TSchema : ISchema
+{
+}
+
+/// <summary>
+/// Handles a WebSocket connection based on the sub-protocol specified.
+/// </summary>
+public interface IWebSocketHandler
+{
+    /// <summary>
+    /// Executes a specified WebSocket request, returning once the connection is closed.
```

Comment:
Do I understand correctly, that once being accepted WebSocket object is kept alive for ages handling all messages flowing between client and server throught it?

Reply:
Yes that is how ASP.NET Core handles WebSocket connections. When the middleware returns, the associated WebSocket connection/object is automatically closed. So the middleware won’t return until the WebSocket connection has disconnected, or until it wants to disconnect.